Fail to build rolling base

Thanks of all, thanks for putting this repository up, it's very helpful to experiment with Nix! I wanted to try to build a ROS2 workspace, and so I typed:
```
nix build ros/20221020-1#rolling.ros_base.ws.contents
```

Which fails with:
```
error: builder for '/nix/store/ssszwq15pba7d9m3phaw3snpy1p2r134-Mimick-a819614.drv' failed with exit code 1;
       last 10 log lines:
       >  * [new branch]      features/travis       -> origin/features/travis
       >  * [new branch]      features/windows      -> origin/features/windows
       >  * [new branch]      master                -> origin/master
       >  * [new branch]      ros2                  -> origin/ros2
       >  * [new tag]         v0.1.0                -> v0.1.0
       >  * [new tag]         v0.2.0                -> v0.2.0
       >  * [new tag]         v0.3.0                -> v0.3.0
       > fatal: git cat-file: could not get object info
       > Unrecognized git object type:
       > Unable to checkout a819614f43592f551697e5bc9dba8a16e7d9f44d from https://github.com/ros2/Mimick.git.
       For full logs, run 'nix log /nix/store/ssszwq15pba7d9m3phaw3snpy1p2r134-Mimick-a819614.drv'.
error: 1 dependencies of derivation '/nix/store/lyj6g003x07nhmxcs0878dzvszdfcmny-mimick_vendor.drv' failed to build
```

It seems like the flake for mimick is attempting to download the wrong git tag. The data in the nix flake seems to point to the correct tag for `mimick_vendor` ( https://github.com/clearpathrobotics/nix-ros/blob/latest/rolling/srcs/mimick_vendor.nix ). However, `mimick_vendor` adds `mimick` as an external project using CMake. Using `nix show-derivation ros#rolling.mimick_vendor` I see that there is an override applied to that package, that uses this store path as the git repository:
```
/nix/store/ssszwq15pba7d9m3phaw3snpy1p2r134-Mimick-a819614.drv

      "postPatch": "sed -i '\\|GIT_REPOSITORY\\s.*'https://github.com/ros2/Mimick.git'|c\\\n  URL \"/nix/store/90lqhsfb4ss99a3g67k3g8x8iqih1lq4-Mimick-a819614\"' \\\n  'CMakeLists.txt'\n",
```

However, the CMakeLists of `mimick_vendor` specifies that we should download Mimick at version `de11f8377eb95f932a03707b583bf3d4ce5bd3e7`. I have been looking around, but since I'm really new with Nix, I can't quite quite figure out just yet how that `a819614` reference got into the system, although it seems like the rolling override is to blame :thinking: 

Thanks again for the great talk at ROSCon!

Actually this little patch fixes the wrong git tag, I didn't know the overrides were manually specified:
```diff
diff --git a/overrides/rolling.nix b/overrides/rolling.nix
index fe42b41..fc7de4a 100644
--- a/overrides/rolling.nix
+++ b/overrides/rolling.nix
@@ -92,8 +92,8 @@ in rosFinal: rosPrev: {
   mimick_vendor = (patchVendorGit rosPrev.mimick_vendor {
     url = "https://github.com/ros2/Mimick.git";
     fetchgitArgs = {
-      rev = "a819614f43592f551697e5bc9dba8a16e7d9f44d";
-      sha256 = "sha256-LHz0xFt7Q1HKPJBDUnQoHEPbWLZ2zBwj4vxy0LZ2c5c=";
+      rev = "de11f8377eb95f932a03707b583bf3d4ce5bd3e7";
+      sha256 = "sha256-adCxIl0F3QkgSimOhvuTyhmig1rFy/K9wxZ/+YCuxYo=";
     };
   });
 ```

However, I am now getting another error....
```
/nix/store/hyz46mc9bnjl6rq6hzbwjaybjybzav8c-glibc-2.35-163-dev/include/features.h:412:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [8;;https://gcc.gnu.org/onlinedocs/gcc/>
  412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
```

But the `flags.txt` seems to correctly specify `-O2` so I need to figure out why `mimick` does not see those flags...

--- a/overrides/rolling.nix
+++ b/overrides/rolling.nix
@@ -89,13 +89,23 @@ in rosFinal: rosPrev: {
     };
   };
 
-  mimick_vendor = (patchVendorGit rosPrev.mimick_vendor {
+  mimick_vendor = patchVendorGit (
+    rosPrev.mimick_vendor.overrideColconAttrs(_: {
+      patches = [
+        # Pending: https://github.com/ros2/mimick_vendor/pull/26
+        (fetchpatch {
+          url = "https://github.com/ros2/mimick_vendor/commit/e57ed638c309a6b51a6b137f8103b7e002f69fec.patch";
+          hash = "sha256-ofpuKyZaWBUgxLC4Cow7NyTPqvW94hdtetSSdMcAWc4=";
+        })
+      ];
+    }))
+  {
     url = "https://github.com/ros2/Mimick.git";
     fetchgitArgs = {
-      rev = "a819614f43592f551697e5bc9dba8a16e7d9f44d";
-      sha256 = "sha256-LHz0xFt7Q1HKPJBDUnQoHEPbWLZ2zBwj4vxy0LZ2c5c=";
+      rev = "de11f8377eb95f932a03707b583bf3d4ce5bd3e7";
+      sha256 = "sha256-adCxIl0F3QkgSimOhvuTyhmig1rFy/K9wxZ/+YCuxYo=";
     };
-  });
+  };
 
   rmw_implementation = rosPrev.rmw_implementation.overrideColconAttrs(old: {
     colconRunDepends = old.colconRunDepends ++ [ rosFinal.rmw_cyclonedds_cpp ];

